mne/preprocessing/xdawn.py

```diff
@@ -73,30 +83,24 @@ def _least_square_evoked(epochs_data, events, tmin, sfreq):
 
     # Compute the independent evoked responses per condition, while correcting
     # for event overlaps.
-    n_min, n_max = int(tmin * sfreq), int(tmax * sfreq)
-    window = n_max - n_min
-    n_samples = raw.shape[1]
-    toeplitz = list()
-    classes = np.unique(events[:, 2])
-    for ii, this_class in enumerate(classes):
-        # select events by type
-        sel = events[:, 2] == this_class
 
-        # build toeplitz matrix
-        trig = np.zeros((n_samples, 1))
-        ix_trig = (events[sel, 0]) + n_min
-        trig[ix_trig] = 1
-        toeplitz.append(linalg.toeplitz(trig[0:window], trig))
```

Comment:
That's being handled inside _prepare_rerp_preds now. The most clearly similar part is in line 355, with `sparse.dia_matrix`.

It used to be `toeplitz` there, too, IIRC, but we went with sparse after a while.

Timing the xdawn test, I get this patch making things a bit slower (~15%) all in all. Haven't tested it on a real use scenario, maybe @kingjr has one?